.github/actions/setup-build-env/action.yml

```diff
@@ -0,0 +1,26 @@
+name: "Setup build environment"
+description: "Setup the runner with all the tools we need to build our project."
+inputs:
+    mock-google-services:
+        description: "Mock the google-services.json file inside the gradle modules."
+        default: "false"
+        required: false
+runs:
+    using: "composite"
+    steps:
+        - name: Mock google-services.json
+          shell: bash
+          if: ${{ inputs.mock-google-services == 'true' }}
+          run: |
+              cp .github/mock-google-services.json app/google-services.json
+              cp .github/mock-google-services.json wear/google-services.json
+              cp .github/mock-google-services.json automotive/google-services.json
+
+        - name: Set up JDK 17
```

Comment:
Not sure about this, but maybe https://github.com/home-assistant/android/pull/5108#discussion_r1993341746 is better changed here? The other PR already has a lot going on and needs a rebase anyway if this one is merged.

Reply:
I don't mind the rebase but if you prefer I can bring the change to this PR.

Reply:
In fact I would prefer to keep it in the other PR since it comes with some explanation in the description of the PR that explain why I did the update.